src/v/cloud_roles/apply_gcp_credentials.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "cloud_roles/apply_credentials.h"
+
+namespace cloud_roles {
+
+class apply_gcp_credentials final
+  : public cloud_roles::apply_credentials::impl {
+public:
+    explicit apply_gcp_credentials(cloud_roles::gcp_credentials&& credentials);
```

Comment:
it is generally preferable to introduce types as they are needed. here `gcp_credentials` is used, for the first time, but it was introduced a few commits back.

Reply:
moved commits around so these types are introduced in the commit where they are used. there are a couple of places where I was not able to follow this but mostly I have made the changes.